### PR TITLE
[stable/prometheus-operator] Improve rules sync rules and description

### LINKED ISSUE #11527: [stable/prometheus-operator] Synced alerts from upstream projects use different selection labels from chart
The stable/prometheus-operator chart has scripts that will automatically sync alerts and dashboards from upstream projects. In some cases it is possible for these alerts to have alternate selectors due to the upstream project having a different deployment process. For example:
```
count_values("config_hash", alertmanager_config_hash{job="{{ $alertmanagerJob }}"}) BY (service) / ON(service) GROUP_LEFT() label_replace(prometheus_operator_spec_replicas{job="{{ $operatorJob }}",controller="alertmanager"}, "service", "alertmanager-$1", "name", "(.*)") != 1
```
[source](https://github.com/helm/charts/blob/master/stable/prometheus-operator/templates/alertmanager/rules/alertmanager.rules.yaml#L26)

This alert uses a relabel expression to select an alertmanager service that does not get deployed by the chart, and so it will never trigger.

---

Creating a completely universal script that is capable of dealing with these conditions may be beyond the scope of this project, however, validating the functionality of the alerts, and amending the configuration to function with the chart is still necessary.

What are your thoughts on this, @paskal @gianrubio ?

### PR DESCRIPTION
**Pull request by paskal**
Hello, team!

This PR:
* Bump kube-state-metrics from version 0.13.1 to version 0.14.1
* Clarify instructions to making changes on Prometheus rules and Grafana dashboards as asked in [this comment](https://github.com/helm/charts/pull/11407#issuecomment-465104838)
* Move PrometheusRules into prometheus folder from alertmanager, as per [this comment](https://github.com/helm/charts/pull/10051#issuecomment-449097153)
* Rename `grafana-coredns-k8s` to `k8s-coredns` to be in line with other files naming
* Adjust `namespace="monitoring"` to be replaced by proper namespace of helm deployment
* Adjust `"alertmanager-$1"` to be replaced by `"$1"`, as per [this comment](https://github.com/helm/charts/issues/11527#issuecomment-466058720), fix #11527
* Sync rules and dashboards

/cc @vsliouniaev, @gianrubio

### CODE REVIEW

**Review by gianrubio — APPROVED** [review]
Really nice contribution @paskal, thanks for all your effort!

/ok-to-test
/lgtm

**Comment by paskal** [comment]
Something was wrong with requirements.lock, fixed now.

File: stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml [hunk 1]
```diff
@@ -25,9 +27,19 @@ spec:
       for: 10m
       labels:
         severity: warning
-    - alert: DeadMansSwitch
+    - alert: Watchdog
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml:30]: I would consider this a major change. 
- People may have (and we do!) custom configurations around this alert that will all need updating. Anything using the old name will break 
- https://github.com/helm/charts/blob/master/stable/prometheus-operator/values.yaml#L108 will need to get changed too

**Review by vsliouniaev — CHANGES_REQUESTED** [review]
Thanks for doing all of this work! I think we should consider a major version bump because of the renaming of DeadManSwitch to Watchdog.
I think if this PR is merged as-is it would count as a fix to #11527

File: stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml (hunk 1 above)
Comment (paskal) [reply stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml:30]: Should be fixed now, see latest commit.

**Comment by paskal** [comment]
Seems like it won't work without `lgtm`.

**Comment by vsliouniaev** [comment]
Gave this another look and it all appears to be working on my cluster
/lgtm

File: stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml (hunk 1 above)
Comment (eduardobaitello) [reply stable/prometheus-operator/templates/prometheus/rules/general.rules.yaml:30]: > * People may have (and we do!) custom configurations around this alert that will all need updating. Anything using the old name will break.

@vsliouniaev Yes, it breaks my [workaround](https://stackoverflow.com/a/54814033/7641078).

Thankfully I read your warning before upgrading :+1:

**Pull request merged**